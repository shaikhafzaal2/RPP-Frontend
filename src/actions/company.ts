import { createAction, PayloadAction } from '@reduxjs/toolkit';

import { ActionTypes } from 'literals';
import { actionPayload } from 'modules/helpers';
import { Company } from 'types/common';

export const getCompanyRequest = createAction(ActionTypes.USER_GET_COMPANY_REQUEST,  (payload: any) => actionPayload(payload),);
export const getCompanySuccess = createAction<PayloadAction<Company>>(ActionTypes.USER_GET_COMPANY_SUCCESS);
export const getCompanyFailure = createAction(ActionTypes.USER_GET_COMPANY_FAILURE,  (payload: any) => actionPayload(payload),);