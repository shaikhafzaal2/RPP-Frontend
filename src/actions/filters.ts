import { createAction, PayloadAction } from '@reduxjs/toolkit';

import { ActionTypes } from 'literals';
import { actionPayload } from 'modules/helpers';

import { Filters,  } from 'types/common';

export const updateFilters = createAction<PayloadAction<Filters>>(ActionTypes.UPDATE_FILTERS);
export const getFiltersRequest = createAction(ActionTypes.GET_FILTER_REQUEST);
export const getFiltersRequestSuccess = createAction(ActionTypes.GET_FILTER_SUCCESS,  (payload: any) => actionPayload(payload),);
export const getFiltersRequestFailure = createAction(ActionTypes.GET_FILTER_FAILURE);