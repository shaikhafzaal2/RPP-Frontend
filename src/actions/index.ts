export * from './alerts';
export * from './app';
export * from './github';
export * from './user';
export * from './company';
export * from './profile';
export * from './filters'