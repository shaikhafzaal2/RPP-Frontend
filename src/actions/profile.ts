import { createAction, PayloadAction } from '@reduxjs/toolkit';

import { ActionTypes } from 'literals';
import { actionPayload } from 'modules/helpers';
import { Profile } from 'types/common';

export const getProfileRequest = createAction(ActionTypes.USER_GET_PROFILE_REQUEST,  (payload: any) => actionPayload(payload),);
export const getProfileSuccess = createAction<PayloadAction<Profile>>(ActionTypes.USER_GET_PROFILE_SUCCESS);
export const getProfileFailure = createAction(ActionTypes.USER_GET_PROFILE_FAILURE,  (payload: any) => actionPayload(payload),)
export const uploadProfileRequest = createAction(ActionTypes.UPLOAD_PROFILE_REQUEST,(payload: any) => actionPayload(payload),);
export const uploadProfileSuccess = createAction(ActionTypes.UPLOAD_PROFILE_SUCCESS);
export const uploadProfileFailure = createAction(ActionTypes.UPLOAD_PROFILE_FAILURE);
