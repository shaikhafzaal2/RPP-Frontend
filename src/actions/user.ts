import { AccountInfo } from '@azure/msal-browser';
import { createAction } from '@reduxjs/toolkit';

import { ActionTypes } from 'literals';
import { actionPayload } from 'modules/helpers';



export const loginSuccess = createAction(ActionTypes.USER_LOGIN_SUCCESS,(payload:AccountInfo) => actionPayload(payload));
export const loginFailure = createAction(ActionTypes.USER_LOGIN_FAILURE,  (payload: any) => actionPayload(payload),);
export const loginRequest = createAction(ActionTypes.USER_LOGIN_REQUEST);
export const logoutRequest = createAction(ActionTypes.USER_LOGOUT_REQUEST);
export const logoutSuccess = createAction(ActionTypes.USER_LOGOUT_SUCCESS);


