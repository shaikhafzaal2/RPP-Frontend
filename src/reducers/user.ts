import { createReducer } from '@reduxjs/toolkit';


import { loginFailure, loginSuccess, logoutSuccess } from 'actions';

import { UserState } from 'types';

export const userState: UserState = {
  isLoggedIn: false,
  user: null,
  error: null,
};
export default {
  user: createReducer<UserState>(userState, (builder) => {
  builder
    .addCase(loginSuccess, (state, action) => {
      state.isLoggedIn = true;
      state.user = action.payload;
      state.error = null;
    })
    .addCase(loginFailure, (state, action) => {
      state.isLoggedIn = false;
      state.user = null;
      state.error = action.payload;
    })
    .addCase(logoutSuccess, (state) => {
      state.isLoggedIn = false;
      state.user = null;
      state.error = null;
    });
})};
