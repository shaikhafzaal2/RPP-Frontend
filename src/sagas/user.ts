import { all, call, put, takeEvery  } from 'redux-saga/effects';

import { ActionTypes } from 'literals';

import { loginFailure, loginSuccess, logoutSuccess } from 'actions';
import { AccountInfo, PublicClientApplication } from "@azure/msal-browser";
import { msalConfig } from 'authConfig';


const msalInstance = new PublicClientApplication (msalConfig);

export function* loginSaga() {
  
  try {
    let myAccount: AccountInfo;
    myAccount = yield call([msalInstance,msalInstance.loginPopup]);
    yield put(loginSuccess(myAccount));
    
  } catch (error) {
    console.log(error)
    yield put(loginFailure(error))
  }
}


export function* logoutSaga() {
  
  try {
    yield call([msalInstance,msalInstance.logoutPopup]);
    yield put(logoutSuccess());
    
  } catch (error) {
    console.log(error)
   
  }
 
}

function* watchLogout() {
  yield takeEvery(ActionTypes.USER_LOGOUT_REQUEST, logoutSaga);
}


function* watchLogin() {
  yield takeEvery(ActionTypes.USER_LOGIN_REQUEST, loginSaga);
}

export default function* root() {
  yield all([
    // takeLatest(ActionTypes.USER_LOGIN_REQUEST, loginSaga),
    watchLogin(),
    watchLogout(),
  ]);
}
