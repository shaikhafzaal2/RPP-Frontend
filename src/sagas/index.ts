import { all, fork } from 'redux-saga/effects';

import github from './github';
import user from './user';
import company from './company'

/**
 * rootSaga
 */
export default function* root() {
  yield all([fork(github), fork(user), fork(company) ]);
}
