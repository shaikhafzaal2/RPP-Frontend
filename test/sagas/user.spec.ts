import { call, put } from 'redux-saga/effects';
import { AccountInfo, PublicClientApplication } from '@azure/msal-browser';
import { loginSaga, logoutSaga } from 'sagas/user';
import { loginFailure, loginSuccess, logoutSuccess } from 'actions';
import { expectSaga } from 'redux-saga-test-plan';
import user from 'sagas/user';


// Mock the MSAL instance
const mockMsalInstance: PublicClientApplication = {
  loginPopup: jest.fn(),
  logoutPopup: jest.fn(),
} as unknown as PublicClientApplication;

describe('loginSaga', () => {
  it('should have the expected watchers', () =>
  expectSaga(user)
    .run({ silenceTimeout: true })
    .then(result => {
      expect(result.toJSON()).toMatchSnapshot();
    }));
  it('should dispatch loginSuccess with account info on successful login', () => {
    const account: AccountInfo = {
      name: 'Test User',
      homeAccountId: '',
      environment: '',
      tenantId: '',
      username: '',
      localAccountId: ''
    };
    const gen = loginSaga();
  (call([mockMsalInstance, mockMsalInstance.loginPopup]));
    // expect(gen.next(account).value).toEqual(put(loginSuccess(account)));
    // console.log("here is res"+JSON.stringify( gen.next(account).value));
    expect(gen.next().done).toBe(false);
  });

//   it('should dispatch loginFailure with error on failed login', () => {
//     const error = new Error('login failed');
//     const gen = loginSaga();
//     expect(gen.next().value).toEqual(call([mockMsalInstance, mockMsalInstance.loginPopup]));
//     expect(gen.throw(error).value).toEqual(put(loginFailure(error)));
//     expect(gen.next().done).toBe(true);
//   });
// });

// describe('logoutSaga', () => {
//   it('should dispatch logoutSuccess on successful logout', () => {
//     const gen = logoutSaga();
//     expect(gen.next().value).toEqual(call([mockMsalInstance, mockMsalInstance.logoutPopup]));
//     expect(gen.next().value).toEqual(put(logoutSuccess()));
//     expect(gen.next().done).toBe(true);
//   });

//   it('should log error on failed logout', () => {
//     const error = new Error('logout failed');
//     const gen = logoutSaga();
//     expect(gen.next().value).toEqual(call([mockMsalInstance, mockMsalInstance.logoutPopup]));
//     expect(gen.throw(error).done).toBe(true);
//   });
});
