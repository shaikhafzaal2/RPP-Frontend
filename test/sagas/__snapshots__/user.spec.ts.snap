// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginSaga should have the expected watchers 1`] = `
Object {
  "fork": Array [
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [
          "USER_LOGIN_REQUEST",
          "@@redux-saga-test-plan/json/function/loginSaga",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/takeEvery",
      },
      "type": "FORK",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [
          "USER_LOGOUT_REQUEST",
          "@@redux-saga-test-plan/json/function/logoutSaga",
        ],
        "context": null,
        "fn": "@@redux-saga-test-plan/json/function/takeEvery",
      },
      "type": "FORK",
    },
  ],
  "take": Array [
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "pattern": "USER_LOGIN_REQUEST",
      },
      "type": "TAKE",
    },
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "pattern": "USER_LOGOUT_REQUEST",
      },
      "type": "TAKE",
    },
  ],
}
`;
